common/persistence/sql/execution.go

```diff
@@ -126,79 +127,100 @@ func (m *sqlExecutionStore) createWorkflowExecutionTx(
 		return nil, err
 	}
 
-	// current workflow record check
-	if row != nil {
-		// current run ID, last write version, current workflow state check
-		switch request.Mode {
-		case p.CreateWorkflowModeBrandNew:
-			return nil, &p.WorkflowExecutionAlreadyStartedError{
-				Msg:              fmt.Sprintf("Workflow execution already running. WorkflowId: %v", row.WorkflowID),
-				StartRequestID:   row.CreateRequestID,
-				RunID:            row.RunID.String(),
-				State:            row.State,
-				Status:           row.Status,
-				LastWriteVersion: row.LastWriteVersion,
+	// current run ID, last write version, current workflow state check
+	currentWorkflowConditionFailed := &p.CurrentWorkflowConditionFailedError{
+		Msg:              "",
+		RequestID:        "",
+		RunID:            "",
+		State:            enumsspb.WORKFLOW_EXECUTION_STATE_UNSPECIFIED,
+		Status:           enumspb.WORKFLOW_EXECUTION_STATUS_UNSPECIFIED,
+		LastWriteVersion: 0,
+	}
+	if currentRow != nil {
+		currentWorkflowConditionFailed = &p.CurrentWorkflowConditionFailedError{
```

Comment:
Not seeing why this assigns a whole new instance here rather than updated the fields on the instance from just above?

Reply:
not a big difference, but not creating a new error will save GC cost.